os/kernel/init/os_start.c

```diff
@@ -828,6 +828,10 @@ void os_start(void)
 	/* Create initial tasks and bring-up the system */
 
 #ifdef CONFIG_PM
+	/* Registeer Special Domains, which are specific to Kernel*/
+	pm_domain_register("PM_IDLE_DOMAIN");
```

Comment:
Please move in pm_init()

Reply:
The API usage here and the defined API return datatype doesn't tally. If we do not need any return value, then define the API as `void pm_domain_register ()`

Reply:
@ewoodev moved it to header, so we don't need it to initialize.
@edwakuwaku we can't change the return type of pm_domain_register to void, because applications needs domain ID to call pm_resume & pm_suspend APIs

Reply:
Then, for places that uses the API but don't need the return value, you should (void) it